### PR TITLE
Fixing fileio tests for windows

### PR DESCRIPTION
**Pull request by pabloem**
r: @aaltay 
I've tested this for windows, and got fileio tests to pass. See golink at go-slash-fiotpemfx

### CODE REVIEW

File: sdks/python/apache_beam/io/fileio.py [hunk 1]
```diff
@@ -129,9 +129,10 @@ def process(self, file_metadata):
                 if isinstance(file_metadata, (str, unicode))
                 else file_metadata)
 
-    if metadata.path.endswith('/') and self._skip_directories:
+    if ((metadata.path.endswith('/') or metadata.path.endswith('\\'))
```
Comment (aaltay) [inline sdks/python/apache_beam/io/fileio.py]: Can you use `os.sep` (https://docs.python.org/3.3/library/os.html?highlight=os.sep#os.sep) ?

File: sdks/python/apache_beam/io/filesystem.py [hunk 2]
```diff
@@ -679,7 +679,7 @@ def match(self, patterns, limits=None):
 
     def _match(pattern, limit):
       """Find all matching paths to the pattern provided."""
-      if pattern.endswith('/'):
+      if pattern.endswith('/') or pattern.endswith('\\'):
```
Comment (aaltay) [inline sdks/python/apache_beam/io/filesystem.py]: Do we other places that needs this change?

File: sdks/python/apache_beam/io/fileio.py (hunk 1 above)
Comment (pabloem) [reply sdks/python/apache_beam/io/fileio.py]: I'd like to check for both, as we may want to access GCS files in windowss, which use `/`.

File: sdks/python/apache_beam/io/filesystem.py (hunk 2 above)
Comment (pabloem) [reply sdks/python/apache_beam/io/filesystem.py]: I didn't find any other checks for `'/'`.

**Pull request merged**